packages/thunderstore-api/src/apiFetch.ts

```diff
@@ -16,15 +16,18 @@ export async function apiFetch2(args: apiFetchArgs) {
   const url = getUrl(args.config, args.path, args.query);
 
   const response = await fetch(url, {
+    method: args.method ? args.method : "GET",
```

Comment:
Don't do this, change the type hint for `apiFetchArgs` to require the type to be provided instead if you want to enforce it.

The best option would probably be to move both the `method` and `body` arguments under the same object and let the caller control any of the normal fetch parameters like so:

```ts
export type apiFetchArgs = {
  config: RequestConfig;
  path: string;
  query?: string;
  request?: Omit<RequestInit, "headers">;
};
```

Then you can just merge it to the fetch call as-is:

```ts
  const response = await fetch(url, {
    ...(args.request ?? {}),
    headers: {
      ...BASE_HEADERS,
      ...getAuthHeaders(args.config),
    },
  });
```

It should be the caller's responsibility to call the function correctly, and typescript's is used to make sure the caller knows what "correctly" means.